app/main.py

```diff
@@ -2,24 +1,70 @@
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.public_name = name
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: object, owner: object) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: Any) -> None:
+        if not isinstance(value, int):
+            raise TypeError("The value should be an integer.")
+        if not self.min_amount <= value <= self.max_amount:
```

Comment:
This is a bug. The condition should be `self.min_amount <= value and value <= self.max_amount`. Otherwise, if `value` is smaller than `self.min_amount` and larger than `self.max_amount` at the same time, it will pass the check, which is not correct.